### ISSUE TITLE
setup:static-content:deploy with locale searches system root

### ISSUE THREAD

**Issue by ghost** [magento/magento2#3592]
When running `setup:static-content:deploy` with locale it searches for the locale in system root, not relative to the installation or current directory.

The command gives an error message if the specified locale is not found in `/i18n/xx_YY.csv`.

For example:

```
$ magento setup:static-content:deploy fi_FI
Requested languages: fi_FI
File "/i18n/fi_FI.csv" does not exist>
```

The error goes away if I copy a file to `/i18n/fi_FI.csv`.

**Comment by mazhalai** [comment]
@rasmuswikman, thank you for reporting, we have created MAGETWO-49991 to investigate and fix.

**Comment by eddielau** [comment]
@rasmuswikman Can you rerun the command with -vvv to show the stack trace?

**Comment by ghost** [comment]
@eddielau 

```
$ php bin/magento setup:static-content:deploy fi_FI -vvv
Requested languages: fi_FI
File "/i18n/fi_FI.csv" does not exist>
#0 /path/to/magento/vendor/magento/framework/File/Csv.php(119): Magento\Framework\File\Csv->getData('/i18n/fi_FI.csv')
#1 /path/to/magento/vendor/magento/framework/Translate.php(378): Magento\Framework\File\Csv->getDataPairs('/i18n/fi_FI.csv')
#2 /path/to/magento/vendor/magento/framework/Translate.php(272): Magento\Framework\Translate->_getFileData('/i18n/fi_FI.csv')
#3 /path/to/magento/vendor/magento/framework/Translate.php(258): Magento\Framework\Translate->loadModuleTranslationByModulesList(Array)
#4 /path/to/magento/vendor/magento/framework/Translate.php(179): Magento\Framework\Translate->_loadModuleTranslation()
#5 /path/to/magento/vendor/magento/module-deploy/Model/Deployer.php(287): Magento\Framework\Translate->loadData('frontend', true)
#6 /path/to/magento/vendor/magento/module-deploy/Model/Deployer.php(132): Magento\Deploy\Model\Deployer->emulateApplicationLocale('fi_FI', 'frontend')
#7 /path/to/magento/vendor/magento/module-deploy/Console/Command/DeployStaticContentCommand.php(120): Magento\Deploy\Model\Deployer->deploy(Object(Magento\Framework\App\ObjectManagerFactory), Array)
#8 /path/to/magento/vendor/symfony/console/Symfony/Component/Console/Command/Command.php(257): Magento\Deploy\Console\Command\DeployStaticContentCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 /path/to/magento/vendor/symfony/console/Symfony/Component/Console/Application.php(874): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /path/to/magento/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(Magento\Deploy\Console\Command\DeployStaticContentCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /path/to/magento/vendor/magento/framework/Console/Cli.php(49): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /path/to/magento/vendor/symfony/console/Symfony/Component/Console/Application.php(126): Magento\Framework\Console\Cli->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /path/to/magento/bin/magento(25): Symfony\Component\Console\Application->run()
#14 {main}
```

**Comment by tdgroot** [comment]
Ran into the same problem and did a bit of investigation. 

It seems like the problem occurs in the file `vendor/magento/framework/Translate.php`.
Go to the method called `_loadModuleTranslation()`. You'll see 2 calls to `loadModuleTranslationByModulesList()`:
- `$this->loadModuleTranslationByModulesList($allModulesExceptCurrent);`
- `$this->loadModuleTranslationByModulesList([$currentModule]);`

Whereas the second call provides an array with the `$currentModule`. So, `$currentModule` is set this way: 
`$currentModule = $this->getControllerModuleName();`. 
In my case, `$this->getControllerModuleName();` returns null. 

That results into magento 2 not resolving the directory name properly. Assuming the path template would be something like: `$PATH_TO_MODULE/i18n/xx_XX.csv`, magento 2 will try to load `/i18n/xx_XX.csv`. 

Hope this could be helpful to the core dev team.

**Comment by Krapulat** [comment]
I have the same problem. Is it solved in develop?

**Comment by toniyecla** [comment]
development workaround:

`ln -s /path_to_magento2/i18n /`

`bin/magento setup:static-content:deploy es_ES`
Requested languages: es_ES
=== frontend -> Magento/blank -> es_ES ===
.......................................................................................................................................................................................................................................................................................................

**Comment by piotrekkaminski** [comment]
This issues should be solved in 2.1. If not working, please open a new issue.
